zephyr/CMakeLists.txt

```diff
@@ -830,6 +830,8 @@ if (NOT CONFIG_COMPILER_INLINE_FUNCTION_OPTION)
 target_compile_options(SOF INTERFACE -fno-inline-functions)
 endif()
 
+target_compile_options(SOF INTERFACE -Wno-address-of-packed-member)
```

Comment:
Its better we keep this, what errors do you see, there may be other/better ways to resolve ?

Reply:
It's mostly stuff like:

```
/zep_workspace/sof/src/include/sof/coherent.h: In function '__coherent_init_thread':
/zep_workspace/sof/src/include/sof/coherent.h:433:19: warning: taking address of packed member of 'struct coherent' may result in an unaligned pointer value [-Waddress-of-packed-member]
  433 |         list_init(&c->list);
```

Reply:
1. no, no idea why `struct coherent` has to be packed, I'd say it doesn't. It was that way from day one of the coherent API
2. we currently have only one used of the coherent API: src/lib/ams.c - is ams used at all, are there plans to use it? Does it really need and benefit from the coherent API? @kfrydryx 
3. removing that warning seems like not the right fix for this. I found this explanation on [1] why this warning is useful:
> When a program accesses a misaligned member of a packed struct, the compiler generates whatever code is necessary to read or write the correct value.
> 
> If the address of a misaligned member is stored in a pointer object, dereferencing that pointer doesn't give the compiler an opportunity to generate that extra code.

[1] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=51628